### PR TITLE
Fix downgrade warning is unhelpful and confusing

### LINKED ISSUE #5097: Downgrade warning is unhelpful and confusing
When a user wants to run against an older version of Kubernetes, but have already run `minikube start` with the default version, we show them the warning, and ignore their request:

`Kubernetes downgrade is not supported, will continue to use v1.15.2`

Instead, we should exit with a compatibility error that tells the user exactly what is going on and how to workaround it. Something like:

```
Error: You have selected version v1.12.0, but your existing "minikube" cluster is
running version v1.15.2. Non-destructive downgrades are not supported, but you
can proceed by selecting one of these options:

* Run `minikube delete -p minikube` to delete the existing cluster
* Run `minikube start -p <new name>` to start a new cluster with a different name
* Run `minikube start --kubernetes-version=v1.15.2` or newer to continue using
this cluster.
```

Thoughts?

### PR DESCRIPTION
**Pull request by phillipahereza**
Fixes #5097

### CODE REVIEW

**Comment by phillipahereza** [comment]
I signed it

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -820,8 +820,14 @@ func validateKubernetesVersions(old *cfg.Config) (string, bool) {
...
 		nv = version.VersionPrefix + ovs.String()
-		out.ErrT(out.Conflict, "Kubernetes downgrade is not supported, will continue to use {{.version}}", out.V{"version": nv})
-		return nv, isUpgrade
+		exit.WithCodeT(exit.Config, `Error: You have selected version {{.new}}, but your existing "minikube" cluster is
+running version {{.old}}. Non-destructive downgrades are not supported, but you
+can proceed by selecting one of these options:
+
+* Run minikube delete -p minikube to delete the existing cluster
+* Run minikube start -p <new name> to start a new cluster with a different name
+* Run minikube start --kubernetes-version=v1.15.2 or newer to continue using this cluster.`, out.V{"new": nvs, "old": ovs})
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Thanks for doing this! Instead of `v1.15.2`, this should be `{{.new}}`.

Comment (phillipahereza) [reply cmd/minikube/cmd/start.go]: sorry had missed that, now fixed

File: cmd/minikube/cmd/start.go [hunk 2]
```diff
@@ -820,8 +820,14 @@ func validateKubernetesVersions(old *cfg.Config) (string, bool) {
 
 	if nvs.LT(ovs) {
 		nv = version.VersionPrefix + ovs.String()
-		out.ErrT(out.Conflict, "Kubernetes downgrade is not supported, will continue to use {{.version}}", out.V{"version": nv})
-		return nv, isUpgrade
+		exit.WithCodeT(exit.Config, `Error: You have selected version {{.new}}, but your existing "minikube" cluster is
+running version {{.old}}. Non-destructive downgrades are not supported, but you
+can proceed by selecting one of these options:
+
+* Run minikube delete -p minikube to delete the existing cluster
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Instead of `-p minikube`, this should use the node name (old.ClusterName?).

**Comment by tstromberg** [comment]
It appears that the CLA check is failing because first commit in this PR used an e-mail address that isn't linked to your GitHub user: 
  (https://github.com/kubernetes/minikube/pull/5155/commits/531999f20d22af112bd1bebfa256b09ae1701c34)

See: https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

https://www.git-tower.com/learn/git/faq/change-author-name-email shows how to change the e-mail address for previous commits.

**Comment by tstromberg** [comment]
travis failure is unrelated (and fixed at head)

**Comment by tstromberg** [comment]
Current output generated by this PR:

```
😄  minikube v1.3.1 on Darwin 10.14.6
💣  Error: You have selected version 1.14.0, but your existing cluster for profile "minikube" is running version 1.16.0-beta.1. Non-destructive downgrades are not supported, but you can proceed by performing one of the following options:

* Run "minikube start [-p minikube] --kubernetes-version=1.16.0-beta.1" or newer to continue using this cluster
* Run "minikube start -p <new name> --kubernetes-version=1.14.0" to start a new cluster with a different name
* Run "minikube delete [-p minikube]" to delete the existing cluster first, and "minikube start [-p minikube] --kubernetes-version=1.14.0" to start a new cluster under the same name
```

File: cmd/minikube/cmd/start.go [hunk 3]
```diff
@@ -820,8 +820,12 @@ func validateKubernetesVersions(old *cfg.Config) (string, bool) {
 
 	if nvs.LT(ovs) {
 		nv = version.VersionPrefix + ovs.String()
-		out.ErrT(out.Conflict, "Kubernetes downgrade is not supported, will continue to use {{.version}}", out.V{"version": nv})
-		return nv, isUpgrade
+		exit.WithCodeT(exit.Config, `Error: You have selected version {{.new}}, but your existing cluster for profile "{{.profile}}" is running version {{.old}}. Non-destructive downgrades are not supported, but you can proceed by performing one of the following options:
+
+* Run "minikube start [-p {{.profile}}] --kubernetes-version={{.old}}" or newer to continue using this cluster
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Remove the [] around the -p arguments here and elsewhere. You should expect that users will cut and paste this command everywhere.

If you wanted to make the output slightly smarter for the usual case where a user has not specified a custom profile name, you could do something like:

profileArg := ""
if  cfg.GetMachineName() != constants.DefaultMachineName {
  profileArg = fmt.Sprintf(" -p %s", cfg.GetMachineName())
}

and then just add profileArg to your exit message.

File: cmd/minikube/cmd/start.go [hunk 4]
```diff
@@ -820,8 +820,12 @@ func validateKubernetesVersions(old *cfg.Config) (string, bool) {
...
+* Run "minikube start -p <new name> --kubernetes-version={{.new}}" to start a new cluster with a different name
+* Run "minikube delete [-p {{.profile}}]" to delete the existing cluster first, and "minikube start [-p {{.profile}}] --kubernetes-version={{.new}}" to start a new cluster under the same name`, out.V{"new": nvs, "old": ovs, "profile": cfg.GetMachineName()})
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Great work! My apologies for not giving you the right advice up front, but after trying it out, I have some suggestions on improving the wording. For the easiest scanning of text, I feel that the beginning of the message should start with the choice outcome, followed with the steps the user must take.

```
* Recreate the cluster using v<old>: Run "minikube delete<profileFlag>", then "minikube ...
* Create a second cluster with v<old>: Run ...
* Reuse the existing cluster with v<new>: Run ...
```

What do you think?

**Comment by tstromberg** [comment]
Thanks for working through this! I'm pleased as punch to see that this is finally being fixed.

**Comment by RA489** [comment]
@minikube-bot OK to test

File: cmd/minikube/cmd/start.go (hunk 4 above)
Comment (phillipahereza) [reply cmd/minikube/cmd/start.go]: Yeah sure, that makes more sense

**Comment by phillipahereza** [comment]
Current output generated by this PR
```
😄  [vtest1] minikube v1.3.1 on Darwin 10.14.5
💣  Error: You have selected Kubernetes v1.14.0, but the existing cluster for your profile is running Kubernetes v1.14.2. Non-destructive downgrades are not supported, but you can proceed by performing one of the following options:

* Recreate the cluster using Kubernetes v1.14.0: Run "minikube delete -p vtest1", then "minikube start -p vtest1 --kubernetes-version=1.14.0"
* Create a second cluster with Kubernetes v1.14.0: Run "minikube start -p <new name> --kubernetes-version=1.14.0"
* Reuse the existing cluster with Kubernetes v1.14.2 or newer: Run "minikube start -p vtest1 --kubernetes-version=1.14.2"
```
@tstromberg I still feel the last option is kinda obvious and should be removed, what do you think?

**Comment by phillipahereza** [comment]
@tstromberg I'm having trouble getting this test to pass locally https://github.com/kubernetes/minikube/blob/86e67d8561b5d708f30a25e0084ada42d61f4a10/pkg/drivers/hyperkit/driver_test.go#L69 and I also don't understand how it can pass since https://github.com/kubernetes/minikube/blob/86e67d8561b5d708f30a25e0084ada42d61f4a10/pkg/drivers/hyperkit/driver.go#L71 doesn't pass a value to `DiskSize`

**Comment by tstromberg** [comment]
> @tstromberg I'm having trouble getting this test to pass locally
> 
> https://github.com/kubernetes/minikube/blob/86e67d8561b5d708f30a25e0084ada42d61f4a10/pkg/drivers/hyperkit/driver_test.go#L69
> and I also don't understand how it can pass since
> 
> https://github.com/kubernetes/minikube/blob/86e67d8561b5d708f30a25e0084ada42d61f4a10/pkg/drivers/hyperkit/driver.go#L71
> doesn't pass a value to `DiskSize`

I've now fixed this at head.

**Comment by tstromberg** [comment]
> Current output generated by this PR
> 
> ```
> 😄  [vtest1] minikube v1.3.1 on Darwin 10.14.5
> 💣  Error: You have selected Kubernetes v1.14.0, but the existing cluster for your profile is running Kubernetes v1.14.2. Non-destructive downgrades are not supported, but you can proceed by performing one of the following options:
> 
> * Recreate the cluster using Kubernetes v1.14.0: Run "minikube delete -p vtest1", then "minikube start -p vtest1 --kubernetes-version=1.14.0"
> * Create a second cluster with Kubernetes v1.14.0: Run "minikube start -p <new name> --kubernetes-version=1.14.0"
> * Reuse the existing cluster with Kubernetes v1.14.2 or newer: Run "minikube start -p vtest1 --kubernetes-version=1.14.2"
> ```
> 
> @tstromberg I still feel the last option is kinda obvious and should be removed, what do you think?

It is normally obvious, but I could think of cases where it may not be so clear, such as if someone sets `minikube config set kubernetes-version v1.10.0`, and then doesn't realize that they have a path to move forwards with.

**Comment by tstromberg** [comment]
This looks great. Thank you so much for your patience!

**Pull request merged**